railties/lib/rails/application.rb

```diff
@@ -590,5 +590,46 @@ def build_request(env)
       def build_middleware
         config.app_middleware + super
       end
+
+      class NonSymbolAccessDeprecatedHash < Hash # :nodoc:
+        def initialize(value = nil)
+          if value.is_a?(Hash)
+            value.each_pair { |k, v| self[k] = v }
+          else
+            super
+          end
+        end
+
+        def []=(key, value)
+          if value.is_a?(Hash)
+            value = self.class.new(value)
+          end
+
+          super(key.to_sym, value)
+        end
+
+        def [](key)
+          super(convert_key(key))
+        end
+
+        def key?(key)
+          super(convert_key(key))
+        end
+
+        private
+
+          def convert_key(key) # :doc:
+            unless key.kind_of?(Symbol)
+              ActiveSupport::Deprecation.warn(<<~MESSAGE.squish)
```

Comment:
Nitpick: Since you're doing a heredoc and squish, you can break up the line below with a newline.

Reply:
Sure, will do.